### PR TITLE
Added case to allow listing with AWS SDK

### PR DESCRIPTION
**Pull request by dbampalikis**
The pull request enables listing files in the sda-s3Proxy when using the AWS go SDK.

Before the introduced change, the proxy did not add the `bucket` to the url when trying to list files with the AWS SDK. Specifically, when using the `ListObjectsV2`, the proxy was attempting to list files using the `username` as `bucket` and therefore failing.

### CODE REVIEW

**Review by aaperis — APPROVED** [review]

**Review by jbygdell — CHANGES_REQUESTED** [review]
What happens if the user tries to list the root i.e. not adding the username to the path?

File: proxy.go [hunk 1]
```diff
@@ -193,7 +193,7 @@ func (p *Proxy) prependBucketToHostPath(r *http.Request) {
 			r.URL.RawQuery = r.URL.RawQuery + "&prefix=" + username + "%2F"
 		}
 		log.Debug("new Raw Query: ", r.URL.RawQuery)
-	} else if r.Method == http.MethodGet && strings.Contains(r.URL.String(), "?location") {
+		} else if r.Method == http.MethodGet && (strings.Contains(r.URL.String(), "?location") || strings.Contains(r.URL.String(), "&prefix")) {
```
[the remark this answers is not in the document]
Comment (dbampalikis) [reply proxy.go:196]: It doesn't seem to be a problem, but the fact that we don't have detailed tests for that is a concern. I have added a card for a test suite, but for now it's only about testing manually any corner case that comes to mind

**Review by aaperis — APPROVED** [review]

**Review by jbygdell — APPROVED** [review]
